packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -0,0 +1,214 @@
+					} else if (
+						[ 'first_child', 'last_child' ].includes(
+							relativePosition
+						)
+					) {
+						const { innerBlocks } = getBlock( props.clientId );
+
+						// Any of the current block's child blocks (with the right block type) qualifies
+						// as an auto-inserted first or last child block, as the block might've been
+						// auto-inserted and then moved around a bit by the user.
+						autoInsertedBlock = innerBlocks.find(
+							( { name } ) => name === block.name
+						);
+					}
+
+					if ( autoInsertedBlock ) {
+						acc[ block.name ] = autoInsertedBlock.clientId;
+					}
+
+					// TOOD: If no auto-inserted block was found in any of its designated locations,
+					// we want to check if it's present elsewhere in the block tree.
+					// If it is, we'd consider it manually inserted and would want to remove the
+					// corresponding toggle from the block inspector panel.
+
+					return acc;
+				}, {} );
+
+			return {
+				blockIndex: getBlockIndex( props.clientId ),
+				innerBlocksLength: getBlock( props.clientId )?.innerBlocks
+					?.length,
+				rootClientId: _rootClientId,
+				autoInsertedBlockClientIds: _autoInsertedBlockClientIds,
+			};
+		},
+		[ autoInsertedBlocksForCurrentBlock, props.blockName, props.clientId ]
+	);
+
+	const { insertBlock, removeBlock } = useDispatch( blockEditorStore );
+
+	if ( ! autoInsertedBlocksForCurrentBlock.length ) {
+		return null;
+	}
+
+	// Group by block namespace (i.e. prefix before the slash).
+	const groupedAutoInsertedBlocks = autoInsertedBlocksForCurrentBlock.reduce(
+		( groups, block ) => {
+			const [ prefix ] = block.name.split( '/' );
+			if ( ! groups[ prefix ] ) {
+				groups[ prefix ] = [];
+			}
+			groups[ prefix ].push( block );
+			return groups;
+		},
+		{}
+	);
+
+	const insertBlockIntoDesignatedLocation = ( block, relativePosition ) => {
+		if ( [ 'before', 'after' ].includes( relativePosition ) ) {
+			insertBlock(
+				block,
+				relativePosition === 'after' ? blockIndex + 1 : blockIndex,
+				rootClientId, // Insert as a child of the current block's parent
+				false
+			);
+		} else if (
+			[ 'first_child', 'last_child' ].includes( relativePosition )
+		) {
+			insertBlock(
+				block,
+				// TODO: It'd be great if insertBlock() would accept negative indices for insertion.
+				relativePosition === 'first_child' ? 0 : innerBlocksLength,
+				props.clientId, // Insert as a child of the current block.
+				false
+			);
+		}
+	};
+
+	return (
+		<InspectorControls>
+			<PanelBody title={ __( 'Plugins' ) } initialOpen={ true }>
+				{ Object.keys( groupedAutoInsertedBlocks ).map( ( vendor ) => {
+					return (
+						<Fragment key={ vendor }>
+							<h3>{ vendor }</h3>
+							{ groupedAutoInsertedBlocks[ vendor ].map(
+								( block ) => {
+									// TODO: Display block icon.
+									// <BlockIcon icon={ block.icon } />
+
+									const checked = Object.keys(
+										autoInsertedBlockClientIds
+									).includes( block.name );
```

Comment:
is this different than the non-array construction?

```js
const checked = block.name in autoInsertedBlockClientIds;
```

Reply:
Ah nice! I knew there had to be a JS-native version of this (but was unaware of the exact syntax) 😬

Reply:
2918ca0960